src/registry.rs

```diff
@@ -58,7 +58,7 @@ impl RegistryCore {
             // the collector_id.
             if desc_id_set.insert(desc.id) {
                 // The set did not have this value present, true is returned.
-                collector_id += desc.id;
+                collector_id ^= desc.id;
```

Comment:
It is ported from [client_golang](https://github.com/prometheus/client_golang/blob/master/prometheus/registry.go#L264).

A collector may collect multiple metrics, thus a collector's id would be the sum of metrics's id. Anyway, our goal here is computing a collector's id based on it's metrics.